lib/cache/cache.go

```diff
@@ -1324,6 +1326,12 @@ func (c *Cache) fetchAndWatch(ctx context.Context, retry retryutils.Retry, timer
 
 	c.notify(c.ctx, Event{Type: WatcherStarted})
 
+	fetchAndApplyDuration := time.Since(fetchAndApplyStart)
+	c.Logger.Debugf("Cache %q fetch and apply took %v", c.Config.target, fetchAndApplyDuration)
+	if fetchAndApplyDuration > time.Second*20 {
+		c.Logger.Warnf("Slow fetch and apply for cache %q took %v.", c.Config.target, fetchAndApplyDuration)
```

Comment:
Structured logging please? (also we probably don't need the warn _and_ the debug)